dashboard/app/helpers/thank_donors_interstitial_helper.rb

```diff
@@ -0,0 +1,13 @@
+module ThankDonorsInterstitialHelper
+  # Determine whether or not to show the thank donors interstitial popup to a user
+  # We only show the interstitial to signed in users on their first login,
+  # and also do not want to show it when it might conflict with other dialogs.
+  def self.show?(user)
+    return false if user.nil?
+    return false unless user.sign_in_count == 1
+    return false if user.age.nil? # Age dialog
+    return false if user && user.teacher? && !user.accepted_latest_terms? # Terms of Service dialog
+
+    return true
```

Comment:
Per offline chat: Extracting reusing helpers for the other dialogs we're trying not to compete with would help here.  At that point, this all might be easier to read if stated in the positive.

```rb
def self.show?(user)
  user&.sign_in_count == 1 &&
    !show_age_dialog(user) &&
    !show_terms_of_service(user)
end
```